tests/report_status_test.go

```diff
@@ -0,0 +1,56 @@
+package nmstate_tests
+
+import (
+	"io/ioutil"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	yaml "github.com/ghodss/yaml"
+
+	nmstate_api_v1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate.io/v1"
+
+	k8s_api_v1 "k8s.io/api/apps/v1"
+	k8s_apimachinery_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8s_client_v1 "k8s.io/client-go/kubernetes/typed/apps/v1"
+)
+
+var _ = Describe("Reporting State", func() {
+	Context("periodically", func() {
+		var (
+			dsClient         k8s_client_v1.DaemonSetInterface
+			nodeNetworkState *nmstate_api_v1.NodeNetworkState
+			nodeName         string
+
+			_ = BeforeEach(func() {
+
+				By("Creating the daemon set to monitor state")
+				manifest, err := ioutil.ReadFile("manifests/examples/state-controller-ds.yaml")
+				Expect(err).ShouldNot(HaveOccurred())
+				var ds k8s_api_v1.DaemonSet
+				err = yaml.Unmarshal(manifest, &ds)
+				Expect(err).ShouldNot(HaveOccurred())
+				dsClient = k8sClientset.AppsV1().DaemonSets(*nmstateNs)
```

Comment:
we can chain this with the Create from the following line

Reply:
dsClient is reused at AfterEach that's why I assign it